src/main/java/com/dotloop/loopit/LoopItController.java

```diff
@@ -109,39 +157,43 @@ public void deleteToken() {
     }
 
     @RequestMapping("/auth/callback")
-    public String callback(HttpServletRequest request, @RequestParam String code, @RequestParam String state) {
-
-        // XSRF protection
-        String csrfToken = csrfTokenRepository.loadToken(request).getToken();
-        if (!state.equals(csrfToken)) {
-            throw new ForbiddenException("csrf failure.");
-        }
-
+    public String callback(HttpServletRequest request, @RequestParam String code) {
         // HTTP Basic Auth
         String authStr = Base64Utils.encodeToString((clientId + ":" + clientSecret).getBytes(Consts.UTF_8));
         try {
             Token token = getToken(code, authStr);
-            TokenStore.save(getUser().getUsername(), token);
+            TokenStore.save(username, token);
         } catch (IOException e) {
             e.printStackTrace();
         }
         return "callback";
     }
 
+    private String getProfiles() throws Exception {
+        Token token = TokenStore.get(username);
+
+        if (token == null) {
+            throw new AccessNotGrantedException();
+        } else {
+            logger.debug("Retrieved token : {}", token);
+
+            return makeRequest("get", "/profile");
+        }
+    }
+
     @ResponseStatus(HttpStatus.FORBIDDEN)
     public class ForbiddenException extends RuntimeException {
         public ForbiddenException(String msg) {
             super(msg);
         }
     }
 
-    private String getAuthorizeUrl(String state) {
+    private String getAuthorizeUrl() {
```

Comment:
@sweetcoco any reason that we removed state?  We recommend that our users implement this in our developer guide, so it might be a good idea to include in the sample app if not too difficult